tools/plugin-toolkit/src/initCmd/index.ts

```diff
@@ -1,69 +1,109 @@
 // (C) 2021 GoodData Corporation
 import { ActionOptions, TargetAppFlavor, TargetBackendType } from "../_base/types";
-import { logInfo } from "../_base/cli/loggers";
+import { logError, logInfo } from "../_base/cli/loggers";
 import { promptBackend, promptFlavor, promptHostname, promptName } from "../_base/cli/prompts";
 import kebabCase from "lodash/kebabCase";
 import * as path from "path";
-import mkdirp from "mkdirp";
+import fse from "fs-extra";
 import tar from "tar";
 import { getDashboardPluginTemplateArchive } from "../dashboard-plugin-template";
 import { readJsonSync, writeAsJsonSync } from "../_base/utils";
+import {
+    backendTypeValidator,
+    flavorValidator,
+    hostnameValidatorFactory,
+    isInputValidationError,
+    pluginNameValidator,
+    validOrDie,
+} from "../_base/cli/validators";
+import { processTigerFiles } from "./processTigerFiles";
 
 type InitCmdActionConfig = {
     backend: TargetBackendType;
     hostname: string;
-    flavor: "ts" | "js";
-    targetDir: string;
+    flavor: TargetAppFlavor;
+    targetDir: string | undefined;
     skipInstall: boolean;
 };
 
-function getHostname(options: ActionOptions): string | undefined {
-    return options.programOpts.hostname;
+function getHostname(backend: TargetBackendType | undefined, options: ActionOptions): string | undefined {
+    const { hostname } = options.programOpts;
+
+    if (!hostname) {
+        return undefined;
+    }
+
+    // do the hostname validation only if the backend type is known at this point
+    if (backend !== undefined) {
+        validOrDie("hostname", hostname, hostnameValidatorFactory(backend));
+    }
+
+    return hostname;
 }
 
 function getBackend(options: ActionOptions): TargetBackendType | undefined {
     const { backend } = options.commandOpts;
 
-    if (backend === "bear" || backend === "tiger") {
-        return backend;
+    if (!backend) {
+        return undefined;
     }
 
-    return undefined;
+    validOrDie("backend", backend, backendTypeValidator);
+
+    return backend as TargetBackendType;
 }
 
 function getFlavor(options: ActionOptions): TargetAppFlavor | undefined {
     const { flavor } = options.commandOpts;
 
-    if (flavor === "ts" || flavor === "js") {
-        return flavor;
+    if (!flavor) {
+        return undefined;
```

Comment:
Do we handle undefined flavor somewhere else? Would not be better to return any default value in here?